{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"stylegan-encoder.ipynb","version":"0.3.2","provenance":[]},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"code","metadata":{"id":"qaMHLF4Rzcze","colab_type":"code","colab":{"base_uri":"https://localhost:8080/","height":106},"outputId":"6b1574da-129f-4229-bd82-e01598642f2c","executionInfo":{"status":"ok","timestamp":1560865037580,"user_tz":-540,"elapsed":3941,"user":{"displayName":"nori 86","photoUrl":"","userId":"17990641330801160498"}}},"source":["!git clone https://github.com/Puzer/stylegan-encoder.git"],"execution_count":2,"outputs":[{"output_type":"stream","text":["Cloning into 'stylegan-encoder'...\n","remote: Enumerating objects: 105, done.\u001b[K\n","remote: Total 105 (delta 0), reused 0 (delta 0), pack-reused 105\u001b[K\n","Receiving objects: 100% (105/105), 10.39 MiB | 16.02 MiB/s, done.\n","Resolving deltas: 100% (28/28), done.\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"dkk80IOs0CeN","colab_type":"code","colab":{"base_uri":"https://localhost:8080/","height":35},"outputId":"d946f91b-3747-4bc3-c7d8-24fb43c4804e","executionInfo":{"status":"ok","timestamp":1560865164297,"user_tz":-540,"elapsed":899,"user":{"displayName":"nori 86","photoUrl":"","userId":"17990641330801160498"}}},"source":["%cd stylegan-encoder"],"execution_count":4,"outputs":[{"output_type":"stream","text":["/content/stylegan-encoder\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"7YXAz2k3z9pc","colab_type":"code","colab":{}},"source":["!mkdir img generated_images latent"],"execution_count":0,"outputs":[]},{"cell_type":"code","metadata":{"id":"8W-W69uQzhDy","colab_type":"code","colab":{"base_uri":"https://localhost:8080/","height":873},"outputId":"5d2f09be-5bad-4fc9-d196-97628d0832d6","executionInfo":{"status":"ok","timestamp":1560865420159,"user_tz":-540,"elapsed":17686,"user":{"displayName":"nori 86","photoUrl":"","userId":"17990641330801160498"}}},"source":["!python encode_images.py img/ generated_images/ latent/"],"execution_count":8,"outputs":[{"output_type":"stream","text":["WARNING: Logging before flag parsing goes to stderr.\n","W0618 13:43:24.269385 140510751160192 deprecation_wrapper.py:119] From /content/stylegan-encoder/dnnlib/tflib/tfutil.py:34: The name tf.Dimension is deprecated. Please use tf.compat.v1.Dimension instead.\n","\n","W0618 13:43:24.270150 140510751160192 deprecation_wrapper.py:119] From /content/stylegan-encoder/dnnlib/tflib/tfutil.py:74: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.\n","\n","W0618 13:43:24.270317 140510751160192 deprecation_wrapper.py:119] From /content/stylegan-encoder/dnnlib/tflib/tfutil.py:128: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.\n","\n","Using TensorFlow backend.\n","W0618 13:43:24.303431 140510751160192 deprecation_wrapper.py:119] From /content/stylegan-encoder/dnnlib/tflib/tfutil.py:97: The name tf.get_default_session is deprecated. Please use tf.compat.v1.get_default_session instead.\n","\n","W0618 13:43:24.303757 140510751160192 deprecation_wrapper.py:119] From /content/stylegan-encoder/dnnlib/tflib/tfutil.py:109: The name tf.set_random_seed is deprecated. Please use tf.compat.v1.set_random_seed instead.\n","\n","2019-06-18 13:43:26.237099: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n","W0618 13:43:27.645150 140510751160192 deprecation.py:323] From <string>:364: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.\n","Instructions for updating:\n","Use tf.where in 2.0, which has the same broadcast rule as np.where\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1356, in _do_call\n","    return fn(*args)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1341, in _run_fn\n","    options, feed_dict, fetch_list, target_list, run_metadata)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1429, in _call_tf_sessionrun\n","    run_metadata)\n","tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot assign a device for operation learnable_dlatents/read: {{node learnable_dlatents/read}}was explicitly assigned to /device:GPU:0 but available devices are [ /job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:XLA_CPU:0 ]. Make sure the device specification refers to a valid device. The requested device appears to be a GPU, but CUDA is not enabled.\n","\t [[learnable_dlatents/read]]\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"encode_images.py\", line 79, in <module>\n","    main()\n","  File \"encode_images.py\", line 52, in main\n","    generator = Generator(Gs_network, args.batch_size, randomize_noise=args.randomize_noise)\n","  File \"/content/stylegan-encoder/encoder/generator_model.py\", line 27, in __init__\n","    structure='fixed')\n","  File \"/content/stylegan-encoder/dnnlib/tflib/network.py\", line 450, in run\n","    mb_out = tf.get_default_session().run(out_expr, dict(zip(in_expr, mb_in)))\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 950, in run\n","    run_metadata_ptr)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1173, in _run\n","    feed_dict_tensor, options, run_metadata)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1350, in _do_run\n","    run_metadata)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/client/session.py\", line 1370, in _do_call\n","    raise type(e)(node_def, op, message)\n","tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot assign a device for operation learnable_dlatents/read: node learnable_dlatents/read (defined at /content/stylegan-encoder/encoder/generator_model.py:15) was explicitly assigned to /device:GPU:0 but available devices are [ /job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:XLA_CPU:0 ]. Make sure the device specification refers to a valid device. The requested device appears to be a GPU, but CUDA is not enabled.\n","\t [[learnable_dlatents/read]]\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"zPlukCbC0Ken","colab_type":"code","colab":{}},"source":[""],"execution_count":0,"outputs":[]}]}